### PR TITLE
3formatfixes

### LINKED ISSUE #50: Method 3 consistency with other methods
First, I'll admit I haven't been a part of all of the recent discussions on the new Method 3 so please forgive me if these points have already been discussed.

Method 3 diverges from methods 0, 1, and 2 in a couple of minor ways that, while not significant, grind my gears lol.

- Methods 0 and 1 do not place a `.` separator between the numalgo and the identifier. Method 2 does but this is to make it so the many sections within the identifier are consistently delineated.
- Methods 0, 1, and 2 use multibase and multicodec to encode values.

I would like to see method 3 remain more consistent with the other methods by:

- Removing the `.` at the beginning of the identifier
- Encoding the digest with multibase and multicodec/multihash, preferably using Base58 as the encoding

### PR DESCRIPTION
**Pull request by TelegramSam**
fixes #50

### CODE REVIEW

**Comment by TelegramSam** [comment]
@dbluhm does this adequately fix your issues?

Note the inclusion of the leading `.` of the method 2 DID string for consistency.

File: core.html [hunk 1]
```diff
@@ -288,24 +288,24 @@ <h3>Method 3: DID Shortening with SHA-256 Hash</h3>
                 DID
                 identifier derived from a SHA-256 hash of the original identifier. The benefit of using Method 3 over Method to is the ability to have smaller size didcomm messages as did:peer2. dids tend to be verbose in nature. Method 3 peer dids can only be used after a peer did method 2 has been exchange with the other party and thus can map the shortened did to the longform one. In order to send a message encrypted with method 3 you first MUST send a discover-feature message (using the method 2 as the `to` field) to make sure that the receiving agent is capable of resolving method 3 dids.</p>
             <pre class="example" class="ABNF" title="ABNF for peer DIDs">
-peer-did-method-3 = "did:peer:3." 64*HEXDIG
+peer-did-method-3 = "did:peer:3" 64*HEXDIG
```
Comment (dbluhm) [inline core.html:291]: I think we want something more along the lines of:

```suggestion
peer-did-method-3 = "did:peer:3" transform encnumbasis
transform = "z"
encnumbasis = 46*BASE58BTC
```

**Review by dbluhm — CHANGES_REQUESTED** [review]

**Review by dhh1128 — APPROVED** [review]
This LGTM. I am in favor of merging, and will do so by end of week unless someone objects.

File: core.html [hunk 2]
```diff
@@ -288,24 +288,26 @@ <h3>Method 3: DID Shortening with SHA-256 Hash</h3>
...
+peer-did-method-3 = "did:peer:3" transform encnumbasis
+transform = "z"
+encnumbasis = 46*BASE58BTC
 </pre>
             <ul>
                 <li>Start with the DID generated using Method 2.</li>
                 <li>Take the SHA-256 hash of the generated DID (excluding the "did:peer:2" prefix).</li>
-                <li>Encode the hash using lowercase hexadecimal encoding.</li>
-                <li>Construct the final Method 3 DID by concatenating the prefix "did:peer:3" with the hex-encoded hash.
+                <li>Encode the hash using the base58btc multibase encoding.</li>
```
Comment (dbluhm) [inline core.html:298]: To clarify, my suggestion in #50 was to use multibase + multicodec; multibase gives us the `z` on the front of the and the `multicodec` gives us two prefix bytes of `0x12 0x20` which indicate that the data is a sha2-256 hash 32 bytes in length. It's unclear whether that is adequately captured in these changes.

**Comment by dhh1128** [comment]
@dbluhm and @TelegramSam : are you ready for this to be merged?

**Comment by dbluhm** [comment]
I don't think this is ready -- I checked the example Sam used to see if that contained the expected multicodec prefix on the off chance that detail was just missed in the steps to create the DID. The example does not include the multicodec prefix. I think we should be consistent and include the prefix. Beyond just consistency, this leaves the door open for other hash algorithms in the future.

**Comment by TelegramSam** [comment]
@dbluhm I've fixed this - do me a favor and verify this with code (using my inputs) to see if you get the same results?

**Comment by dbluhm** [comment]
Verified with this quick program:

```python
import sys
from multiformats import multibase, multihash

value = sys.argv[1]
digest = multihash.digest(value.encode(), "sha2-256")
encoded = multibase.encode(digest, "base58btc")
print(encoded)
```

Looks good to me. Thanks!

**Review by dbluhm — APPROVED** [review]

**Comment by dhh1128** [comment]
I'm trying to merge, but github is giving me an unexplained error. I'll check back later and try again; maybe their web server has indigestion.

**Comment by dhh1128** [comment]
Doesn't work a few minutes later on a different browser, either... :-(

**Comment by dhh1128** [comment]
Still not working, an hour later.

**Comment by dhh1128** [comment]
Hooray, it finally merged!

**Pull request merged**